src/Controllers/ExampleController.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+
+namespace App\Controllers;
+
+
+use App\Models\ExampleModel;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Slim\Http\Response;
+
+class ExampleController
+{
+    protected ExampleModel $model;
+
+    public function __construct(ExampleModel $model)
```

Comment:
is this where the magical dependency injection happens?

Reply:
if yes, maybe we can add a comment above e.g.
```
// here, the parameter is automatically supplied by the Dependency Injection Container based on the type hint
```

Reply:
yes it is. its called autowiring

Reply:
cool that's what i thought 😎 in that case i think we should add a comment like that which denotes it (so that students and future colleagues recognise what's happening)